extensions/notebook/src/book/bookTocManager.ts

```diff
@@ -0,0 +1,136 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as path from 'path';
+import { BookTreeItem } from './bookTreeItem';
+import * as yaml from 'js-yaml';
+import * as fs from 'fs';
+import { IJupyterBookSectionV1, IJupyterBookSectionV2, JupyterBookSection } from '../contracts/content';
+import { BookVersion } from './bookModel';
+import * as vscode from 'vscode';
+
+export interface IBookTocManager {
+	updateBook(section: BookTreeItem, updatedBook: BookTreeItem): Promise<void>;
+	createBook(bookContentPath: string, contentFolder: string): Promise<void>
+}
+const allowedFileExtensions: string[] = ['.md', '.ipynb'];
+
+export enum tocSectionOperation {
+	Remove,
+	Add
+}
+
+export class BookTocManager implements IBookTocManager {
+
+	constructor() {
+	}
+
+	async getAllFiles(toc: IJupyterBookSectionV2[], directory: string, filesInDir: string[]): Promise<IJupyterBookSectionV2[]> {
```

Comment:
Why does this only take in v2 sections?

Reply:
For now, we'll support creating books for only v2 books.